## Download data

In [3]:
!wget -r \
--no-directories --no-parent --reject "index.html*" \
-P . \
-A _5x.bedgraph \
https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/031520-TG-bs/Pact_tg/dedup/

--2020-05-04 09:54:56--  https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/031520-TG-bs/Pact_tg/dedup/
Resolving gannet.fish.washington.edu... 128.95.149.52
Connecting to gannet.fish.washington.edu|128.95.149.52|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/html]
Saving to: ‘./index.html’

index.html              [ <=>                ]  42.11K  --.-KB/s    in 0.001s  

2020-05-04 09:54:57 (48.7 MB/s) - ‘./index.html’ saved [43123]

Loading robots.txt; please ignore errors.
--2020-05-04 09:54:57--  https://gannet.fish.washington.edu/robots.txt
Reusing existing connection to gannet.fish.washington.edu:443.
HTTP request sent, awaiting response... 404 Not Found
2020-05-04 09:54:57 ERROR 404: Not Found.

Removing ./index.html since it should be rejected.

--2020-05-04 09:54:57--  https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/031520-TG-bs/Pact_tg/dedup/?C=N;O=D
Reusing existing connection to gannet.fish.washington.edu:443.

In [4]:
!wget -r \
--no-directories --no-parent --reject "index.html*" \
-P . \
-A _5x.bedgraph \
https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/031520-TG-bs/Pact_tg/nodedup/

--2020-05-04 09:55:35--  https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/031520-TG-bs/Pact_tg/nodedup/
Resolving gannet.fish.washington.edu... 128.95.149.52
Connecting to gannet.fish.washington.edu|128.95.149.52|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/html]
Saving to: ‘./index.html’

index.html              [ <=>                ]  19.51K  --.-KB/s    in 0s      

2020-05-04 09:55:35 (51.4 MB/s) - ‘./index.html’ saved [19983]

Loading robots.txt; please ignore errors.
--2020-05-04 09:55:35--  https://gannet.fish.washington.edu/robots.txt
Reusing existing connection to gannet.fish.washington.edu:443.
HTTP request sent, awaiting response... 404 Not Found
2020-05-04 09:55:35 ERROR 404: Not Found.

Removing ./index.html since it should be rejected.

--2020-05-04 09:55:35--  https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/031520-TG-bs/Pact_tg/nodedup/?C=N;O=D
Reusing existing connection to gannet.fish.washington.edu:

In [5]:
!ls

Meth1_R1_001_val_1_bismark_bt2_pe._5x.bedgraph
Meth2_R1_001_val_1_bismark_bt2_pe._5x.bedgraph
Meth3_R1_001_val_1_bismark_bt2_pe._5x.bedgraph
Meth4_R1_001_val_1_bismark_bt2_pe._5x.bedgraph
Meth5_R1_001_val_1_bismark_bt2_pe._5x.bedgraph
Meth6_R1_001_val_1_bismark_bt2_pe._5x.bedgraph
Meth7_R1_001_val_1_bismark_bt2_pe._5x.bedgraph
Meth8_R1_001_val_1_bismark_bt2_pe._5x.bedgraph
Meth9_R1_001_val_1_bismark_bt2_pe._5x.bedgraph
gannet.fish.washington.edu


In [6]:
!mkdir Pact

In [7]:
!mv Meth* Pact/

In [8]:
!mkdir Mcap

In [9]:
cd Mcap/

/Users/strigg/Documents/GitHub/Meth_Compare/data/Mcap


In [10]:
!wget -r \
--no-directories --no-parent --reject "index.html*" \
-P . \
-A _5x.bedgraph \
https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/031520-TG-bs/Mcap_tg/nodedup/

--2020-05-04 09:57:07--  https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/031520-TG-bs/Mcap_tg/nodedup/
Resolving gannet.fish.washington.edu... 128.95.149.52
Connecting to gannet.fish.washington.edu|128.95.149.52|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/html]
Saving to: ‘./index.html’

index.html              [ <=>                ]  19.31K  --.-KB/s    in 0s      

2020-05-04 09:57:08 (51.8 MB/s) - ‘./index.html’ saved [19778]

Loading robots.txt; please ignore errors.
--2020-05-04 09:57:08--  https://gannet.fish.washington.edu/robots.txt
Reusing existing connection to gannet.fish.washington.edu:443.
HTTP request sent, awaiting response... 404 Not Found
2020-05-04 09:57:08 ERROR 404: Not Found.

Removing ./index.html since it should be rejected.

--2020-05-04 09:57:08--  https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/031520-TG-bs/Mcap_tg/nodedup/?C=N;O=D
Reusing existing connection to gannet.fish.washington.edu:

In [11]:
!wget -r \
--no-directories --no-parent --reject "index.html*" \
-P . \
-A _5x.bedgraph \
https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/031520-TG-bs/Mcap_tg/dedup/

--2020-05-04 09:57:26--  https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/031520-TG-bs/Mcap_tg/dedup/
Resolving gannet.fish.washington.edu... 128.95.149.52
Connecting to gannet.fish.washington.edu|128.95.149.52|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/html]
Saving to: ‘./index.html’

index.html              [ <=>                ]  42.27K  --.-KB/s    in 0.001s  

2020-05-04 09:57:27 (52.7 MB/s) - ‘./index.html’ saved [43285]

Loading robots.txt; please ignore errors.
--2020-05-04 09:57:27--  https://gannet.fish.washington.edu/robots.txt
Reusing existing connection to gannet.fish.washington.edu:443.
HTTP request sent, awaiting response... 404 Not Found
2020-05-04 09:57:27 ERROR 404: Not Found.

Removing ./index.html since it should be rejected.

--2020-05-04 09:57:27--  https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/031520-TG-bs/Mcap_tg/dedup/?C=N;O=D
Reusing existing connection to gannet.fish.washington.edu:443.

In [ ]:
#check MD5s

In [ ]:
#get md5 hashes for Mcap files

In [12]:
!md5 *.bedgraph

MD5 (Meth10_R1_001_val_1_bismark_bt2_pe._5x.bedgraph) = 83035e7e47b8ad486de22dacc17ae8ed
MD5 (Meth11_R1_001_val_1_bismark_bt2_pe._5x.bedgraph) = 6493359aad0b4228f65b5e563d337ceb
MD5 (Meth12_R1_001_val_1_bismark_bt2_pe._5x.bedgraph) = b2f097299df0cb7d518d22338fdcf39f
MD5 (Meth13_R1_001_val_1_bismark_bt2_pe._5x.bedgraph) = 2bb476cb98072f0e76bfb5c318246c38
MD5 (Meth14_R1_001_val_1_bismark_bt2_pe._5x.bedgraph) = fc0f66cf04ffebe76d61c1db75cfed6e
MD5 (Meth15_R1_001_val_1_bismark_bt2_pe._5x.bedgraph) = 1d7c24b238dc72cd92346213b3523611
MD5 (Meth16_R1_001_val_1_bismark_bt2_pe._5x.bedgraph) = 04fb72d5df60656e6cec15637164fbec
MD5 (Meth17_R1_001_val_1_bismark_bt2_pe._5x.bedgraph) = 073d1c40116a3f93f7a7022cfb4cd3d2
MD5 (Meth18_R1_001_val_1_bismark_bt2_pe._5x.bedgraph) = a255210553db073e5458ccb523a34798


In [ ]:
#get md5 hashes for Pact files

In [13]:
!md5 ../Pact/*.bedgraph

MD5 (../Pact/Meth1_R1_001_val_1_bismark_bt2_pe._5x.bedgraph) = d634ffc3f062d248e36b8dddc9a315e0
MD5 (../Pact/Meth2_R1_001_val_1_bismark_bt2_pe._5x.bedgraph) = 1ec934f5b4ce012b64b77dd69d70ee5f
MD5 (../Pact/Meth3_R1_001_val_1_bismark_bt2_pe._5x.bedgraph) = c456156b7f6a11543d8dc697e8e74b4e
MD5 (../Pact/Meth4_R1_001_val_1_bismark_bt2_pe._5x.bedgraph) = 736fd3802ce1b45b6eb32abf6e1bcb3f
MD5 (../Pact/Meth5_R1_001_val_1_bismark_bt2_pe._5x.bedgraph) = ed4428a6c8cb6a4964687d91c0d8ccb3
MD5 (../Pact/Meth6_R1_001_val_1_bismark_bt2_pe._5x.bedgraph) = 5994ba73d412d8992f2465b148f5ae80
MD5 (../Pact/Meth7_R1_001_val_1_bismark_bt2_pe._5x.bedgraph) = a2b842c439c3df3fb699690cd5b55d5a
MD5 (../Pact/Meth8_R1_001_val_1_bismark_bt2_pe._5x.bedgraph) = c838562956c7abe3656a2b7438a40dc1
MD5 (../Pact/Meth9_R1_001_val_1_bismark_bt2_pe._5x.bedgraph) = c9a4b002113e2501d81e4762cf952b79


In [14]:
cd ../

/Users/strigg/Documents/GitHub/Meth_Compare/data


## Pact data

### Part 1: Merge sample bedgraphs by method

In [ ]:
#merge bed files for RRBS samples

In [15]:
!cat \
Pact/Meth4_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Pact/Meth5_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Pact/Meth6_R1_001_val_1_bismark_bt2_pe._5x.bedgraph |\
/Users/Shared/bioinformatics/bedtools2/bin/sortBed \
-i stdin |\
/Users/Shared/bioinformatics/bedtools2/bin/mergeBed \
-d -1 \
-i stdin \
> Pact/Pact_RRBS_5x.bed


In [16]:
!head Pact/Pact_RRBS_5x.bed

scaffold100000_cov51	109	111
scaffold100000_cov51	112	114
scaffold100000_cov51	205	207
scaffold100000_cov51	213	215
scaffold100000_cov51	236	238
scaffold100004_cov43	100	102
scaffold100004_cov43	107	109
scaffold100009_cov142	180	182
scaffold100009_cov142	212	214
scaffold100017_cov107	1005	1007


In [17]:
!wc -l Pact/Meth4_R1_001_val_1_bismark_bt2_pe._5x.bedgraph 
!wc -l Pact/Meth5_R1_001_val_1_bismark_bt2_pe._5x.bedgraph
!wc -l Pact/Meth6_R1_001_val_1_bismark_bt2_pe._5x.bedgraph 


 1835561 Pact/Meth4_R1_001_val_1_bismark_bt2_pe._5x.bedgraph
 1451229 Pact/Meth5_R1_001_val_1_bismark_bt2_pe._5x.bedgraph
 1517358 Pact/Meth6_R1_001_val_1_bismark_bt2_pe._5x.bedgraph


In [18]:
!wc -l Pact/Pact_RRBS_5x.bed

 2265779 Pact/Pact_RRBS_5x.bed


#### QC check 
awk sort uniq should give the same number of loci as bedtools merge

In [21]:
!cat \
Pact/Meth4_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Pact/Meth5_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Pact/Meth6_R1_001_val_1_bismark_bt2_pe._5x.bedgraph |\
awk -F"\t" '{print $1"\t"$2"\t"$3}' |\
head -3

scaffold6_cov64	2536	2538
scaffold6_cov64	2584	2586
scaffold6_cov64	2676	2678
awk: write error on /dev/stdout
 input record number 1106, file 
 source line number 1
cat: stdout: Broken pipe


In [22]:
!cat \
Pact/Meth4_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Pact/Meth5_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Pact/Meth6_R1_001_val_1_bismark_bt2_pe._5x.bedgraph |\
awk -F"\t" '{print $1"\t"$2"\t"$3}' |\
sort |\
uniq \
> Pact/Pact_RRBS_5x.QCbed

In [23]:
!wc -l Pact/Pact_RRBS_5x.QCbed

 2265779 Pact/Pact_RRBS_5x.QCbed


QC is good since line counts match for:

- 2265779 Pact/Pact_RRBS_5x.bed
- 2265779 Pact/Pact_RRBS_5x.QCbed


In [ ]:
#merge bed files for WGBS samples

In [24]:
!cat \
Pact/Meth1_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Pact/Meth2_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Pact/Meth3_R1_001_val_1_bismark_bt2_pe._5x.bedgraph |\
/Users/Shared/bioinformatics/bedtools2/bin/sortBed \
-i stdin |\
/Users/Shared/bioinformatics/bedtools2/bin/mergeBed \
-d -1 \
-i stdin \
> Pact/Pact_WGBS_5x.bed


In [25]:
!wc -l Pact/Meth1_R1_001_val_1_bismark_bt2_pe._5x.bedgraph 
!wc -l Pact/Meth2_R1_001_val_1_bismark_bt2_pe._5x.bedgraph 
!wc -l Pact/Meth3_R1_001_val_1_bismark_bt2_pe._5x.bedgraph 


 5546051 Pact/Meth1_R1_001_val_1_bismark_bt2_pe._5x.bedgraph
 6358722 Pact/Meth2_R1_001_val_1_bismark_bt2_pe._5x.bedgraph
 5866786 Pact/Meth3_R1_001_val_1_bismark_bt2_pe._5x.bedgraph


In [26]:
!wc -l Pact/Pact_WGBS_5x.bed

 7096944 Pact/Pact_WGBS_5x.bed


#### QC check

In [27]:
!cat \
Pact/Meth1_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Pact/Meth2_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Pact/Meth3_R1_001_val_1_bismark_bt2_pe._5x.bedgraph |\
awk -F"\t" '{print $1"\t"$2"\t"$3}' |\
sort |\
uniq \
> Pact/Pact_WGBS_5x.QCbed

In [29]:
!wc -l Pact/Pact_WGBS_5x.QCbed

 7096944 Pact/Pact_WGBS_5x.QCbed


QC is good since line counts match for:
    
- 7096944 Pact/Pact_WGBS_5x.bed
- 7096944 Pact/Pact_WGBS_5x.QCbed

In [ ]:
#merge bed files for MBD-BS samples

In [30]:
!cat \
Pact/Meth7_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Pact/Meth8_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Pact/Meth9_R1_001_val_1_bismark_bt2_pe._5x.bedgraph |\
/Users/Shared/bioinformatics/bedtools2/bin/sortBed \
-i stdin |\
/Users/Shared/bioinformatics/bedtools2/bin/mergeBed \
-d -1 \
-i stdin \
> Pact/Pact_MBD_5x.bed


In [31]:
!wc -l Pact/Meth7_R1_001_val_1_bismark_bt2_pe._5x.bedgraph 
!wc -l Pact/Meth8_R1_001_val_1_bismark_bt2_pe._5x.bedgraph 
!wc -l Pact/Meth9_R1_001_val_1_bismark_bt2_pe._5x.bedgraph 


 2640625 Pact/Meth7_R1_001_val_1_bismark_bt2_pe._5x.bedgraph
  539008 Pact/Meth8_R1_001_val_1_bismark_bt2_pe._5x.bedgraph
 2732607 Pact/Meth9_R1_001_val_1_bismark_bt2_pe._5x.bedgraph


In [32]:
!wc -l Pact/Pact_MBD_5x.bed

 3704036 Pact/Pact_MBD_5x.bed


#### QC check

In [33]:
!cat \
Pact/Meth7_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Pact/Meth8_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Pact/Meth9_R1_001_val_1_bismark_bt2_pe._5x.bedgraph |\
awk -F"\t" '{print $1"\t"$2"\t"$3}' |\
sort |\
uniq \
> Pact/Pact_MBD_5x.QCbed

In [34]:
!wc -l Pact/Pact_MBD_5x.QCbed

 3704036 Pact/Pact_MBD_5x.QCbed


QC is good since line count matches for:
    
- 3704036 Pact/Pact_MBD_5x.bed
- 3704036 Pact/Pact_MBD_5x.QCbed

### Part 2: Generate genome CpG bedgraph

In [ ]:
#get genome CpG data

In [35]:
cd Pact

/Users/strigg/Documents/GitHub/Meth_Compare/data/Pact


In [36]:
!wget \
https://gannet.fish.washington.edu/seashell/bu-github/Meth_Compare/genome-feature-files/Pact_CpG.gff

--2020-05-04 11:40:12--  https://gannet.fish.washington.edu/seashell/bu-github/Meth_Compare/genome-feature-files/Pact_CpG.gff
Resolving gannet.fish.washington.edu... 128.95.149.52
Connecting to gannet.fish.washington.edu|128.95.149.52|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1111983287 (1.0G)
Saving to: ‘Pact_CpG.gff’

Pact_CpG.gff        100%[===================>]   1.04G  70.8MB/s    in 15s     

2020-05-04 11:40:28 (70.2 MB/s) - ‘Pact_CpG.gff’ saved [1111983287/1111983287]



In [ ]:
#create bedgraph file from CpG.gff

In [37]:
!awk -F"\t" '{print $1"\t"$4"\t"$5}' Pact_CpG.gff |\
/Users/Shared/bioinformatics/bedtools2/bin/sortBed \
-i stdin |\
awk -F"\t" '{print $0"\t"1}' \
> Pact_CpG.bedgraph

In [38]:
!head Pact_CpG.bedgraph

scaffold100000_cov51	5	6	1
scaffold100000_cov51	23	24	1
scaffold100000_cov51	61	62	1
scaffold100000_cov51	69	70	1
scaffold100000_cov51	79	80	1
scaffold100000_cov51	110	111	1
scaffold100000_cov51	113	114	1
scaffold100000_cov51	206	207	1
scaffold100000_cov51	214	215	1
scaffold100000_cov51	237	238	1


### Part 3:  Reformat method CpG bedgraphs
Bedtools mergeBed only outputs 3 columns (chromosome, start, end)
and removes column 4 from the original bedgraphs.

So I need to add a 4th column back to these bed files in order to run Bedtools unionBed. 

Also, the genome CpG gff and bedgraph coordinates have only 1 basepair between them where the merged 5x bedgraphs have 2 basepairs between. So I need to offset these by adding 1 to the start position.

In [ ]:
#loop through all 5x.bed files I generated 
# add 1 to the start position of each loci (line)
# and add a 4th column containing 1 

In [39]:
%%bash 

for f in *5x.bed
do
awk -F"\t" '{print $1"\t"$2+1"\t"$3"\t"1}' \
${f} \
> $(basename ${f%.bed}).bedgraph
done

### Part 4:  Generate union bedgraph 

In [ ]:
#create union bed file

In [42]:
!/Users/Shared/bioinformatics/bedtools2/bin/unionBedGraphs \
-header \
-filler N/A \
-names ALL WGBS RRBS MBD \
-i \
Pact_CpG.bedgraph \
Pact_WGBS_5x.bedgraph \
Pact_RRBS_5x.bedgraph \
Pact_MBD_5x.bedgraph \
> Pact_method_union_5x.bedgraph

In [46]:
!head Pact_method_union_5x.bedgraph

chrom	start	end	ALL	WGBS	RRBS	MBD
scaffold100000_cov51	5	6	1	N/A	N/A	N/A
scaffold100000_cov51	23	24	1	1	N/A	N/A
scaffold100000_cov51	61	62	1	1	N/A	N/A
scaffold100000_cov51	69	70	1	1	N/A	N/A
scaffold100000_cov51	79	80	1	1	N/A	N/A
scaffold100000_cov51	110	111	1	1	1	N/A
scaffold100000_cov51	113	114	1	1	1	N/A
scaffold100000_cov51	206	207	1	1	1	N/A
scaffold100000_cov51	214	215	1	1	1	N/A


### Part 5: Create input table for Upset plotting in R using union bedgraph

In [43]:
#replace NA with zeros and
#combine fields 1-3 into 1 string that will serve as rowname
!cat Pact_method_union_5x.bedgraph |\
sed 's/N\/A/0/g' |\
awk -F"\t" '{print$1"_"$2"_"$3"\t"$4"\t"$5"\t"$6"\t"$7}' \
> Pact_method_union_5x_matrix.tsv

In [ ]:
#preview file

In [44]:
!head Pact_method_union_5x_matrix.tsv

chrom_start_end	ALL	WGBS	RRBS	MBD
scaffold100000_cov51_5_6	1	0	0	0
scaffold100000_cov51_23_24	1	1	0	0
scaffold100000_cov51_61_62	1	1	0	0
scaffold100000_cov51_69_70	1	1	0	0
scaffold100000_cov51_79_80	1	1	0	0
scaffold100000_cov51_110_111	1	1	1	0
scaffold100000_cov51_113_114	1	1	1	0
scaffold100000_cov51_206_207	1	1	1	0
scaffold100000_cov51_214_215	1	1	1	0


In [45]:
!wc -l Pact_method_union_5x_matrix.tsv

 9155621 Pact_method_union_5x_matrix.tsv


### Part 6: parse out CpGs from different method combinations for tracks and feature analysis

In [ ]:
#CpGs commonly detected across all methods

In [47]:
!awk -F"\t" '{if(($4==1)&&($5==1)&&($6==1)&&($7==1))print $0}' \
Pact_method_union_5x.bedgraph \
> Pact_WGBS_RRBS_MBD_only_CpGs.bed


In [48]:
#CpGs detected by MBD and WGBS only 

In [61]:
!awk -F"\t" '{if(($4==1)&&($5==1)&&($6=="N\/A")&&($7==1))print $0}' \
Pact_method_union_5x.bedgraph \
> Pact_WGBS_MBD_only_CpGs.bed

In [64]:
#CpGs detected by to MBD and RRBS only

In [65]:
!awk -F"\t" '{if(($4==1)&&($5=="N\/A")&&($6==1)&&($7==1))print $0}' \
Pact_method_union_5x.bedgraph \
> Pact_RRBS_MBD_only_CpGs.bed

In [66]:
#CpGs detected by WGBS and RRBS only

In [67]:
!awk -F"\t" '{if(($4==1)&&($5==1)&&($6==1)&&($7=="N\/A"))print $0}' \
Pact_method_union_5x.bedgraph \
> Pact_WGBS_RRBS_only_CpGs.bed

In [68]:
#CpGs detected by WGBS only

In [69]:
!awk -F"\t" '{if(($4==1)&&($5==1)&&($6=="N\/A")&&($7=="N\/A"))print $0}' \
Pact_method_union_5x.bedgraph \
> Pact_WGBS_only_CpGs.bed

In [70]:
#CpGs detected by RRBS only

In [71]:
!awk -F"\t" '{if(($4==1)&&($5=="N\/A")&&($6==1)&&($7=="N\/A"))print $0}' \
Pact_method_union_5x.bedgraph \
> Pact_RRBS_only_CpGs.bed

In [72]:
#CpGs detected by MBD only

In [73]:
!awk -F"\t" '{if(($4==1)&&($5=="N\/A")&&($6=="N\/A")&&($7==1))print $0}' \
Pact_method_union_5x.bedgraph \
> Pact_MBD_only_CpGs.bed

In [ ]:
#CpGs not detected by any method

In [75]:
!awk -F"\t" '{if(($4==1)&&($5=="N\/A")&&($6=="N\/A")&&($7=="N\/A"))print $0}' \
Pact_method_union_5x.bedgraph \
> Pact_no_method_CpGs.bed

#### QC check: 
all line counts from individual bed files should add up to lines in union bed file

In [76]:
!wc -l *_CpGs.bed

   70784 Pact_MBD_only_CpGs.bed
   10526 Pact_RRBS_MBD_only_CpGs.bed
  148043 Pact_RRBS_only_CpGs.bed
 2360477 Pact_WGBS_MBD_only_CpGs.bed
 1262249 Pact_WGBS_RRBS_MBD_only_CpGs.bed
  844961 Pact_WGBS_RRBS_only_CpGs.bed
 2629257 Pact_WGBS_only_CpGs.bed
 1829323 Pact_no_method_CpGs.bed
 9155620 total


QC is good line counts in union file match the total:
- 9155621 Pact_method_union_5x_matrix.tsv (contains header)

## Mcap data

### Part 1: Merge sample bedgraphs by method

In [77]:
cd ../

/Users/strigg/Documents/GitHub/Meth_Compare/data


In [ ]:
#merge bed files for RRBS samples

In [78]:
!cat \
Mcap/Meth13_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Mcap/Meth14_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Mcap/Meth15_R1_001_val_1_bismark_bt2_pe._5x.bedgraph |\
/Users/Shared/bioinformatics/bedtools2/bin/sortBed \
-i stdin |\
/Users/Shared/bioinformatics/bedtools2/bin/mergeBed \
-d -1 \
-i stdin \
> Mcap/Mcap_RRBS_5x.bed


In [79]:
!head Mcap/Mcap_RRBS_5x.bed

1	3493	3495
1	3518	3520
1	3727	3729
1	3752	3754
1	3757	3759
1	3770	3772
1	11876	11878
1	11887	11889
1	11894	11896
1	11941	11943


In [80]:
!wc -l Mcap/Meth13_R1_001_val_1_bismark_bt2_pe._5x.bedgraph 
!wc -l Mcap/Meth14_R1_001_val_1_bismark_bt2_pe._5x.bedgraph
!wc -l Mcap/Meth15_R1_001_val_1_bismark_bt2_pe._5x.bedgraph 


 3173254 Mcap/Meth13_R1_001_val_1_bismark_bt2_pe._5x.bedgraph
 2648697 Mcap/Meth14_R1_001_val_1_bismark_bt2_pe._5x.bedgraph
 3176517 Mcap/Meth15_R1_001_val_1_bismark_bt2_pe._5x.bedgraph


In [81]:
!wc -l Mcap/Mcap_RRBS_5x.bed

 3981450 Mcap/Mcap_RRBS_5x.bed


#### QC check

In [ ]:
#check if awk sort uniq gives the same number of loci as bedtools merge

In [82]:
!cat \
Mcap/Meth13_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Mcap/Meth14_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Mcap/Meth15_R1_001_val_1_bismark_bt2_pe._5x.bedgraph |\
awk -F"\t" '{print $1"\t"$2"\t"$3}' |\
head -3

1	3493	3495
1	3518	3520
1	3727	3729
awk: write error on /dev/stdout
 input record number 2106, file 
 source line number 1
cat: stdout: Broken pipe


In [84]:
!cat \
Mcap/Meth13_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Mcap/Meth14_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Mcap/Meth15_R1_001_val_1_bismark_bt2_pe._5x.bedgraph |\
awk -F"\t" '{print $1"\t"$2"\t"$3}' |\
sort |\
uniq \
> Mcap/Mcap_RRBS_5x.QCbed

In [85]:
!wc -l Mcap/Mcap_RRBS_5x.QCbed

 3981450 Mcap/Mcap_RRBS_5x.QCbed


QC is good, line counts in files match:
    
- 3981450 Mcap/Mcap_RRBS_5x.bed
- 3981450 Mcap/Mcap_RRBS_5x.QCbed

In [ ]:
#merge bed files for WGBS samples

In [86]:
!cat \
Mcap/Meth10_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Mcap/Meth11_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Mcap/Meth12_R1_001_val_1_bismark_bt2_pe._5x.bedgraph |\
/Users/Shared/bioinformatics/bedtools2/bin/sortBed \
-i stdin |\
/Users/Shared/bioinformatics/bedtools2/bin/mergeBed \
-d -1 \
-i stdin \
> Mcap/Mcap_WGBS_5x.bed


In [87]:
!wc -l Mcap/Meth10_R1_001_val_1_bismark_bt2_pe._5x.bedgraph 
!wc -l Mcap/Meth11_R1_001_val_1_bismark_bt2_pe._5x.bedgraph 
!wc -l Mcap/Meth12_R1_001_val_1_bismark_bt2_pe._5x.bedgraph 


 4571288 Mcap/Meth10_R1_001_val_1_bismark_bt2_pe._5x.bedgraph
 4661716 Mcap/Meth11_R1_001_val_1_bismark_bt2_pe._5x.bedgraph
 8791700 Mcap/Meth12_R1_001_val_1_bismark_bt2_pe._5x.bedgraph


In [88]:
!wc -l Mcap/Mcap_WGBS_5x.bed

 11509837 Mcap/Mcap_WGBS_5x.bed


#### QC check

In [89]:
!cat \
Mcap/Meth10_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Mcap/Meth11_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Mcap/Meth12_R1_001_val_1_bismark_bt2_pe._5x.bedgraph |\
awk -F"\t" '{print $1"\t"$2"\t"$3}' |\
sort |\
uniq \
> Mcap/Mcap_WGBS_5x.QCbed

In [90]:
!wc -l Mcap/Mcap_WGBS_5x.QCbed

 11509837 Mcap/Mcap_WGBS_5x.QCbed


QC check is good, line counts match for files:

- 11509837 Mcap/Mcap_WGBS_5x.bed
- 11509837 Mcap/Mcap_WGBS_5x.QCbed

In [91]:
#merge bed files for MBD-BS samples

In [92]:
!cat \
Mcap/Meth16_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Mcap/Meth17_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Mcap/Meth18_R1_001_val_1_bismark_bt2_pe._5x.bedgraph |\
/Users/Shared/bioinformatics/bedtools2/bin/sortBed \
-i stdin |\
/Users/Shared/bioinformatics/bedtools2/bin/mergeBed \
-d -1 \
-i stdin \
> Mcap/Mcap_MBD_5x.bed


In [93]:
!wc -l Mcap/Meth16_R1_001_val_1_bismark_bt2_pe._5x.bedgraph 
!wc -l Mcap/Meth17_R1_001_val_1_bismark_bt2_pe._5x.bedgraph 
!wc -l Mcap/Meth18_R1_001_val_1_bismark_bt2_pe._5x.bedgraph 


  583599 Mcap/Meth16_R1_001_val_1_bismark_bt2_pe._5x.bedgraph
  242390 Mcap/Meth17_R1_001_val_1_bismark_bt2_pe._5x.bedgraph
  153392 Mcap/Meth18_R1_001_val_1_bismark_bt2_pe._5x.bedgraph


In [94]:
!wc -l Mcap/Mcap_MBD_5x.bed

  866555 Mcap/Mcap_MBD_5x.bed


#### QC check

In [95]:
!cat \
Mcap/Meth16_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Mcap/Meth17_R1_001_val_1_bismark_bt2_pe._5x.bedgraph \
Mcap/Meth18_R1_001_val_1_bismark_bt2_pe._5x.bedgraph |\
awk -F"\t" '{print $1"\t"$2"\t"$3}' |\
sort |\
uniq \
> Mcap/Mcap_MBD_5x.QCbed

In [96]:
!wc -l Mcap/Mcap_MBD_5x.QCbed

  866555 Mcap/Mcap_MBD_5x.QCbed


QC is good, line counts match in files:
    
- 866555 Mcap/Mcap_MBD_5x.bed
- 866555 Mcap/Mcap_MBD_5x.QCbed

### Part 2: Generate genome CpG bedgraph

In [97]:
#get genome CpG data

In [98]:
cd Mcap

/Users/strigg/Documents/GitHub/Meth_Compare/data/Mcap


In [99]:
!wget \
https://gannet.fish.washington.edu/seashell/bu-github/Meth_Compare/genome-feature-files/Mcap_CpG.gff

--2020-05-04 15:15:42--  https://gannet.fish.washington.edu/seashell/bu-github/Meth_Compare/genome-feature-files/Mcap_CpG.gff
Resolving gannet.fish.washington.edu... 128.95.149.52
Connecting to gannet.fish.washington.edu|128.95.149.52|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 2635935327 (2.5G)
Saving to: ‘Mcap_CpG.gff’

Mcap_CpG.gff        100%[===================>]   2.45G  71.7MB/s    in 35s     

2020-05-04 15:16:18 (71.3 MB/s) - ‘Mcap_CpG.gff’ saved [2635935327/2635935327]



In [100]:
#create bedgraph file from CpG.gff

In [101]:
!awk -F"\t" '{print $1"\t"$4"\t"$5}' Mcap_CpG.gff |\
/Users/Shared/bioinformatics/bedtools2/bin/sortBed \
-i stdin |\
awk -F"\t" '{print $0"\t"1}' \
> Mcap_CpG.bedgraph

In [102]:
!head Mcap_CpG.bedgraph

1	37	38	1
1	90	91	1
1	121	122	1
1	132	133	1
1	153	154	1
1	170	171	1
1	220	221	1
1	224	225	1
1	233	234	1
1	272	273	1


### Part 3:  Reformat method CpG bedgraphs

Bedtools mergeBed only outputs 3 columns (chromosome, start, end)
and removes column 4 from the original bedgraphs.

So I need to add a 4th column back to these bed files in order to run Bedtools unionBed. 

Also, the genome CpG gff and bedgraph coordinates have only 1 basepair between them where the merged 5x bedgraphs have 2 basepairs between. So I need to offset these by adding 1 to the start position.



In [ ]:
#loop through all 5x.bed files I generated 
# add 1 to the start position of each loci (line)
# and add a 4th column containing 1 

In [104]:
%%bash 

for f in *5x.bed
do
awk -F"\t" '{print $1"\t"$2+1"\t"$3"\t"1}' \
${f} \
> $(basename ${f%.bed}).bedgraph
done

### Part 4:  Generate union bedgraph 

In [105]:
#create union bed file

In [106]:
!/Users/Shared/bioinformatics/bedtools2/bin/unionBedGraphs \
-header \
-filler N/A \
-names ALL WGBS RRBS MBD \
-i \
Mcap_CpG.bedgraph \
Mcap_WGBS_5x.bedgraph \
Mcap_RRBS_5x.bedgraph \
Mcap_MBD_5x.bedgraph \
> Mcap_method_union_5x.bedgraph

In [107]:
!head Mcap_method_union_5x.bedgraph

chrom	start	end	ALL	WGBS	RRBS	MBD
1	37	38	1	N/A	N/A	N/A
1	90	91	1	N/A	N/A	N/A
1	121	122	1	N/A	N/A	N/A
1	132	133	1	N/A	N/A	N/A
1	153	154	1	N/A	N/A	N/A
1	170	171	1	N/A	N/A	N/A
1	220	221	1	N/A	N/A	N/A
1	224	225	1	N/A	N/A	N/A
1	233	234	1	N/A	N/A	N/A


### Part 5: Create input table for Upset plotting in R using union bedgraph

In [109]:
#replace NA with zeros and
#combine fields 1-3 into 1 string that will serve as rowname
!cat Mcap_method_union_5x.bedgraph |\
sed 's/N\/A/0/g' |\
awk -F"\t" '{print$1"_"$2"_"$3"\t"$4"\t"$5"\t"$6"\t"$7}' \
> Mcap_method_union_5x_matrix.tsv

In [110]:
#preview file

In [111]:
!head Mcap_method_union_5x_matrix.tsv

chrom_start_end	ALL	WGBS	RRBS	MBD
1_37_38	1	0	0	0
1_90_91	1	0	0	0
1_121_122	1	0	0	0
1_132_133	1	0	0	0
1_153_154	1	0	0	0
1_170_171	1	0	0	0
1_220_221	1	0	0	0
1_224_225	1	0	0	0
1_233_234	1	0	0	0


In [112]:
!wc -l Mcap_method_union_5x_matrix.tsv

 28675391 Mcap_method_union_5x_matrix.tsv


### Part 6: parse out CpGs from different method combinations for tracks and feature analysis

In [114]:
#CpGs commonly detected across all methods

In [115]:
!awk -F"\t" '{if(($4==1)&&($5==1)&&($6==1)&&($7==1))print $0}' \
Mcap_method_union_5x.bedgraph \
> Mcap_WGBS_RRBS_MBD_only_CpGs.bed


In [116]:
#CpGs detected by MBD and WGBS only 

In [117]:
!awk -F"\t" '{if(($4==1)&&($5==1)&&($6=="N\/A")&&($7==1))print $0}' \
Mcap_method_union_5x.bedgraph \
> Mcap_WGBS_MBD_only_CpGs.bed

In [118]:
#CpGs detected by to MBD and RRBS only

In [119]:
!awk -F"\t" '{if(($4==1)&&($5=="N\/A")&&($6==1)&&($7==1))print $0}' \
Mcap_method_union_5x.bedgraph \
> Mcap_RRBS_MBD_only_CpGs.bed

In [120]:
#CpGs detected by WGBS and RRBS only

In [121]:
!awk -F"\t" '{if(($4==1)&&($5==1)&&($6==1)&&($7=="N\/A"))print $0}' \
Mcap_method_union_5x.bedgraph \
> Mcap_WGBS_RRBS_only_CpGs.bed

In [122]:
#CpGs detected by WGBS only

In [123]:
!awk -F"\t" '{if(($4==1)&&($5==1)&&($6=="N\/A")&&($7=="N\/A"))print $0}' \
Mcap_method_union_5x.bedgraph \
> Mcap_WGBS_only_CpGs.bed

In [124]:
#CpGs detected by RRBS only

In [125]:
!awk -F"\t" '{if(($4==1)&&($5=="N\/A")&&($6==1)&&($7=="N\/A"))print $0}' \
Mcap_method_union_5x.bedgraph \
> Mcap_RRBS_only_CpGs.bed

In [126]:
#CpGs detected by MBD only

In [127]:
!awk -F"\t" '{if(($4==1)&&($5=="N\/A")&&($6=="N\/A")&&($7==1))print $0}' \
Mcap_method_union_5x.bedgraph \
> Mcap_MBD_only_CpGs.bed

In [128]:
#CpGs not detected by any method

In [129]:
!awk -F"\t" '{if(($4==1)&&($5=="N\/A")&&($6=="N\/A")&&($7=="N\/A"))print $0}' \
Mcap_method_union_5x.bedgraph \
> Mcap_no_method_CpGs.bed

#### QC check: 
all line counts from individual bed files should add up to lines in union bed file

In [130]:
!wc -l *_CpGs.bed

   88218 Mcap_MBD_only_CpGs.bed
   19873 Mcap_RRBS_MBD_only_CpGs.bed
 1722340 Mcap_RRBS_only_CpGs.bed
  577695 Mcap_WGBS_MBD_only_CpGs.bed
  180769 Mcap_WGBS_RRBS_MBD_only_CpGs.bed
 2058468 Mcap_WGBS_RRBS_only_CpGs.bed
 8692905 Mcap_WGBS_only_CpGs.bed
 15335122 Mcap_no_method_CpGs.bed
 28675390 total


In [ ]:
QC check is good, line count in union file matches the total:

- 28675391 Mcap_method_union_5x_matrix.tsv  (contains header)

In [ ]:
## Copy all output to Gannet

In [131]:
mkdir /Volumes/web/metacarcinus/FROGER_meth_compare/20200504

!rsync \
--progress \
--verbose \
--archive \
--exclude "gannet.fish.washington.edu" \
/Users/strigg/Documents/GitHub/Meth_Compare/data \
/Volumes/web/metacarcinus/FROGER_meth_compare/20200504

### Output lives here:

[https://gannet.fish.washington.edu/metacarcinus/FROGER_meth_compare/20200504/data/](https://gannet.fish.washington.edu/metacarcinus/FROGER_meth_compare/20200504/data/)

****

## 2020-05-01 
#### below is preliminary code to generate Upset plot input data
output from code is here:  
[https://gannet.fish.washington.edu/metacarcinus/FROGER_meth_compare/20200501/data/](https://gannet.fish.washington.edu/metacarcinus/FROGER_meth_compare/20200501/data/)

In [ ]:
#get data

In [2]:
cd ../data

/Users/strigg/Documents/GitHub/Meth_Compare/data


In [2]:
!wget -r \
--no-directories --no-parent --reject "index.html*" \
-P . \
-A 5x.tab https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/031520-TG-bs/Pact_tg/dedup/

--2020-05-01 11:27:45--  https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/031520-TG-bs/Pact_tg/dedup/
Resolving gannet.fish.washington.edu... 128.95.149.52
Connecting to gannet.fish.washington.edu|128.95.149.52|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/html]
Saving to: ‘./index.html’

index.html              [ <=>                ]  42.11K  --.-KB/s    in 0.001s  

2020-05-01 11:27:46 (42.1 MB/s) - ‘./index.html’ saved [43123]

Loading robots.txt; please ignore errors.
--2020-05-01 11:27:46--  https://gannet.fish.washington.edu/robots.txt
Reusing existing connection to gannet.fish.washington.edu:443.
HTTP request sent, awaiting response... 404 Not Found
2020-05-01 11:27:46 ERROR 404: Not Found.

Removing ./index.html since it should be rejected.

--2020-05-01 11:27:46--  https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/031520-TG-bs/Pact_tg/dedup/?C=N;O=D
Reusing existing connection to gannet.fish.washington.edu:443.

In [3]:
!wget -r \
--no-directories --no-parent --reject "index.html*" \
-P . \
-A 5x.tab https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/031520-TG-bs/Pact_tg/nodedup/

--2020-05-01 11:28:27--  https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/031520-TG-bs/Pact_tg/nodedup/
Resolving gannet.fish.washington.edu... 128.95.149.52
Connecting to gannet.fish.washington.edu|128.95.149.52|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/html]
Saving to: ‘./index.html’

index.html              [ <=>                ]  19.51K  --.-KB/s    in 0s      

2020-05-01 11:28:28 (50.7 MB/s) - ‘./index.html’ saved [19983]

Loading robots.txt; please ignore errors.
--2020-05-01 11:28:28--  https://gannet.fish.washington.edu/robots.txt
Reusing existing connection to gannet.fish.washington.edu:443.
HTTP request sent, awaiting response... 404 Not Found
2020-05-01 11:28:28 ERROR 404: Not Found.

Removing ./index.html since it should be rejected.

--2020-05-01 11:28:28--  https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/031520-TG-bs/Pact_tg/nodedup/?C=N;O=D
Reusing existing connection to gannet.fish.washington.edu:

In [4]:
!ls

Meth1_R1_001_val_1_bismark_bt2_pe._5x.tab
Meth2_R1_001_val_1_bismark_bt2_pe._5x.tab
Meth3_R1_001_val_1_bismark_bt2_pe._5x.tab
Meth4_R1_001_val_1_bismark_bt2_pe._5x.tab
Meth5_R1_001_val_1_bismark_bt2_pe._5x.tab
Meth6_R1_001_val_1_bismark_bt2_pe._5x.tab
Meth7_R1_001_val_1_bismark_bt2_pe._5x.tab
Meth8_R1_001_val_1_bismark_bt2_pe._5x.tab
Meth9_R1_001_val_1_bismark_bt2_pe._5x.tab
gannet.fish.washington.edu


In [5]:
!md5 *.tab

MD5 (Meth1_R1_001_val_1_bismark_bt2_pe._5x.tab) = b2fe2251349d1a1fd2064555711e5f3c
MD5 (Meth2_R1_001_val_1_bismark_bt2_pe._5x.tab) = a54138a247461dfa388ecb7c80a3f757
MD5 (Meth3_R1_001_val_1_bismark_bt2_pe._5x.tab) = 269ee6acbf8bda68699073ed1a8eea11
MD5 (Meth4_R1_001_val_1_bismark_bt2_pe._5x.tab) = c9c7710fe93f163d0563719721334d7d
MD5 (Meth5_R1_001_val_1_bismark_bt2_pe._5x.tab) = b83df95995f0006e49760299d6dce230
MD5 (Meth6_R1_001_val_1_bismark_bt2_pe._5x.tab) = a98b640f20d4d5e216777274ed6760c9
MD5 (Meth7_R1_001_val_1_bismark_bt2_pe._5x.tab) = 18975dcf8f19e868e70fa295179a124d
MD5 (Meth8_R1_001_val_1_bismark_bt2_pe._5x.tab) = b2c9124297fc036d62f744ac2f1b793b
MD5 (Meth9_R1_001_val_1_bismark_bt2_pe._5x.tab) = 81a70ca77975e17ff050c41bbbf46c3f


In [ ]:
# create bed files with total CpG counts

In [6]:
%%bash 

for f in *_5x.tab
do
awk -F"\t" '{print $1"\t"$2"\t"$3"\t"$5+$6}' \
${f} \
> $(basename ${f%_5x.tab})_5x.tC.bedgraph
done

In [7]:
!ls

Meth1_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph
Meth1_R1_001_val_1_bismark_bt2_pe._5x.tab
Meth2_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph
Meth2_R1_001_val_1_bismark_bt2_pe._5x.tab
Meth3_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph
Meth3_R1_001_val_1_bismark_bt2_pe._5x.tab
Meth4_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph
Meth4_R1_001_val_1_bismark_bt2_pe._5x.tab
Meth5_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph
Meth5_R1_001_val_1_bismark_bt2_pe._5x.tab
Meth6_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph
Meth6_R1_001_val_1_bismark_bt2_pe._5x.tab
Meth7_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph
Meth7_R1_001_val_1_bismark_bt2_pe._5x.tab
Meth8_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph
Meth8_R1_001_val_1_bismark_bt2_pe._5x.tab
Meth9_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph
Meth9_R1_001_val_1_bismark_bt2_pe._5x.tab
gannet.fish.washington.edu


In [8]:
!head -5 Meth1_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph

scaffold1_cov55	102	104	6
scaffold1_cov55	105	107	6
scaffold1_cov55	116	118	5
scaffold1_cov55	119	121	5
scaffold1_cov55	146	148	5


In [ ]:
#merge bed files for RRBS samples

In [9]:
!cat \
Meth4_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph \
Meth5_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph \
Meth5_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph |\
/Users/Shared/bioinformatics/bedtools2/bin/sortBed \
-i stdin |\
/Users/Shared/bioinformatics/bedtools2/bin/mergeBed \
-d -1 \
-i stdin \
> RRBS_5x.tc.bed


In [10]:
!head RRBS_5x.tc.bed

scaffold100000_cov51	109	111
scaffold100000_cov51	112	114
scaffold100000_cov51	205	207
scaffold100000_cov51	213	215
scaffold100000_cov51	236	238
scaffold100004_cov43	100	102
scaffold100004_cov43	107	109
scaffold100009_cov142	180	182
scaffold100009_cov142	212	214
scaffold100017_cov107	1052	1054


In [11]:
!wc -l RRBS_5x.tc.bed

 2102829 RRBS_5x.tc.bed


In [12]:
!wc -l Meth4_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph 
!wc -l Meth5_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph 
!wc -l Meth5_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph 


 1835561 Meth4_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph
 1451229 Meth5_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph
 1451229 Meth5_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph


In [13]:
!head -5 Meth4_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph 
!head -5 Meth5_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph 
!head -5 Meth5_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph 

scaffold6_cov64	2536	2538	5
scaffold6_cov64	2584	2586	5
scaffold6_cov64	2676	2678	5
scaffold6_cov64	4588	4590	8
scaffold6_cov64	5101	5103	5
scaffold6_cov64	2584	2586	5
scaffold6_cov64	2676	2678	9
scaffold6_cov64	2682	2684	5
scaffold6_cov64	4553	4555	5
scaffold6_cov64	4588	4590	11
scaffold6_cov64	2584	2586	5
scaffold6_cov64	2676	2678	9
scaffold6_cov64	2682	2684	5
scaffold6_cov64	4553	4555	5
scaffold6_cov64	4588	4590	11


In [ ]:
#merge WGBS samples

In [14]:
!cat \
Meth1_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph \
Meth2_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph \
Meth3_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph |\
/Users/Shared/bioinformatics/bedtools2/bin/sortBed \
-i stdin |\
/Users/Shared/bioinformatics/bedtools2/bin/mergeBed \
-d -1 \
-i stdin \
> WGBS_5x.tc.bed

In [ ]:
#merge MBD samples

In [15]:
!cat \
Meth7_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph \
Meth8_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph \
Meth9_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph |\
/Users/Shared/bioinformatics/bedtools2/bin/sortBed \
-i stdin |\
/Users/Shared/bioinformatics/bedtools2/bin/mergeBed \
-d -1 \
-i stdin \
> MBD_5x.tc.bed

In [ ]:
#get genome CpG data

In [16]:
!wget \
https://gannet.fish.washington.edu/seashell/bu-github/Meth_Compare/genome-feature-files/Pact_CpG.gff

--2020-05-01 11:48:37--  https://gannet.fish.washington.edu/seashell/bu-github/Meth_Compare/genome-feature-files/Pact_CpG.gff
Resolving gannet.fish.washington.edu... 128.95.149.52
Connecting to gannet.fish.washington.edu|128.95.149.52|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1111983287 (1.0G)
Saving to: ‘Pact_CpG.gff’

Pact_CpG.gff        100%[===================>]   1.04G  67.2MB/s    in 15s     

2020-05-01 11:48:53 (69.3 MB/s) - ‘Pact_CpG.gff’ saved [1111983287/1111983287]



In [ ]:
#create bedgraph file from CpG.gff

In [38]:
!awk -F"\t" '{print $1"\t"$4"\t"$5}' Pact_CpG.gff |\
/Users/Shared/bioinformatics/bedtools2/bin/sortBed \
-i stdin |\
awk -F"\t" '{print $0"\t"1}' \
> Pact_CpG.bedgraph

In [40]:
!head Pact_CpG.bedgraph

scaffold100000_cov51	5	6	1
scaffold100000_cov51	23	24	1
scaffold100000_cov51	61	62	1
scaffold100000_cov51	69	70	1
scaffold100000_cov51	79	80	1
scaffold100000_cov51	110	111	1
scaffold100000_cov51	113	114	1
scaffold100000_cov51	206	207	1
scaffold100000_cov51	214	215	1
scaffold100000_cov51	237	238	1


In [ ]:
#merging bed files removed last column so I need to create a 4th column

In [26]:
!ls

MBD_5x.tc.bed
Meth1_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph
Meth1_R1_001_val_1_bismark_bt2_pe._5x.tab
Meth2_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph
Meth2_R1_001_val_1_bismark_bt2_pe._5x.tab
Meth3_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph
Meth3_R1_001_val_1_bismark_bt2_pe._5x.tab
Meth4_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph
Meth4_R1_001_val_1_bismark_bt2_pe._5x.tab
Meth5_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph
Meth5_R1_001_val_1_bismark_bt2_pe._5x.tab
Meth6_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph
Meth6_R1_001_val_1_bismark_bt2_pe._5x.tab
Meth7_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph
Meth7_R1_001_val_1_bismark_bt2_pe._5x.tab
Meth8_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph
Meth8_R1_001_val_1_bismark_bt2_pe._5x.tab
Meth9_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph
Meth9_R1_001_val_1_bismark_bt2_pe._5x.tab
Pact_CpG.bed
Pact_CpG.gff
Pact_method_union_5x.bedgraph
RRBS_5x.tc.bed
WGBS_5x.tc.bed
gannet.fish.washington.edu


In [27]:
%%bash 

for f in *.bed
do
awk -F"\t" '{print $0"\t"1}' \
${f} \
> $(basename ${f%.bed}).bedgraph
done

In [29]:
!head MBD_5x.tc.bedgraph

scaffold100003_cov99	76	78	1
scaffold100003_cov99	97	99	1
scaffold100003_cov99	111	113	1
scaffold100003_cov99	145	147	1
scaffold100003_cov99	176	178	1
scaffold100003_cov99	230	232	1
scaffold100003_cov99	256	258	1
scaffold100003_cov99	285	287	1
scaffold100003_cov99	901	903	1
scaffold100003_cov99	913	915	1


In [31]:
!/Users/Shared/bioinformatics/bedtools2/bin/unionBedGraphs \
-header \
-filler N/A \
-names ALL WGBS RRBS MBD \
-i \
Pact_CpG.bedgraph \
WGBS_5x.tc.bedgraph \
RRBS_5x.tc.bedgraph \
MBD_5x.tc.bedgraph |\
head



chrom	start	end	ALL	WGBS	RRBS	MBD
scaffold100000_cov51	5	6	1	N/A	N/A	N/A
scaffold100000_cov51	22	23	N/A	1	N/A	N/A
scaffold100000_cov51	23	24	1	1	N/A	N/A
scaffold100000_cov51	60	61	N/A	1	N/A	N/A
scaffold100000_cov51	61	62	1	1	N/A	N/A
scaffold100000_cov51	68	69	N/A	1	N/A	N/A
scaffold100000_cov51	69	70	1	1	N/A	N/A
scaffold100000_cov51	78	79	N/A	1	N/A	N/A
scaffold100000_cov51	79	80	1	1	N/A	N/A


In [ ]:
#oddly WGBS has loci that aren't in the genome CpG file;
#This seems like it could be the case because the merged intervals from Bismark 
#are actually 3 bp instead of 2

In [32]:
!awk '{if($1=="scaffold100000_cov51")print $0}' WGBS_5x.tc.bedgraph | head

scaffold100000_cov51	22	24	1
scaffold100000_cov51	60	62	1
scaffold100000_cov51	68	70	1
scaffold100000_cov51	78	80	1
scaffold100000_cov51	109	111	1
scaffold100000_cov51	112	114	1
scaffold100000_cov51	205	207	1
scaffold100000_cov51	213	215	1
scaffold100000_cov51	236	238	1
scaffold100000_cov51	245	247	1


In [ ]:
# i need to adjust the starting coordinate for all data because it shouldn't count twice

In [36]:
%%bash 

for f in *5x.tc.bed
do
awk -F"\t" '{print $1"\t"$2+1"\t"$3"\t"1}' \
${f} \
> $(basename ${f%.bed}).bedgraph
done

In [39]:
!awk '{if($1=="scaffold100000_cov51")print $0}' WGBS_5x.tc.bedgraph | head

scaffold100000_cov51	23	24	1
scaffold100000_cov51	61	62	1
scaffold100000_cov51	69	70	1
scaffold100000_cov51	79	80	1
scaffold100000_cov51	110	111	1
scaffold100000_cov51	113	114	1
scaffold100000_cov51	206	207	1
scaffold100000_cov51	214	215	1
scaffold100000_cov51	237	238	1
scaffold100000_cov51	246	247	1


In [ ]:
#create union table

In [41]:
!/Users/Shared/bioinformatics/bedtools2/bin/unionBedGraphs \
-header \
-filler N/A \
-names ALL WGBS RRBS MBD \
-i \
Pact_CpG.bedgraph \
WGBS_5x.tc.bedgraph \
RRBS_5x.tc.bedgraph \
MBD_5x.tc.bedgraph |\
head

chrom	start	end	ALL	WGBS	RRBS	MBD
scaffold100000_cov51	5	6	1	N/A	N/A	N/A
scaffold100000_cov51	23	24	1	1	N/A	N/A
scaffold100000_cov51	61	62	1	1	N/A	N/A
scaffold100000_cov51	69	70	1	1	N/A	N/A
scaffold100000_cov51	79	80	1	1	N/A	N/A
scaffold100000_cov51	110	111	1	1	1	N/A
scaffold100000_cov51	113	114	1	1	1	N/A
scaffold100000_cov51	206	207	1	1	1	N/A
scaffold100000_cov51	214	215	1	1	1	N/A


In [42]:
!/Users/Shared/bioinformatics/bedtools2/bin/unionBedGraphs \
-header \
-filler N/A \
-names ALL WGBS RRBS MBD \
-i \
Pact_CpG.bedgraph \
WGBS_5x.tc.bedgraph \
RRBS_5x.tc.bedgraph \
MBD_5x.tc.bedgraph \
> Pact_method_union_5x.bedgraph

In [43]:
!tail Pact_method_union_5x.bedgraph

scaffold9_cov118	2310	2311	1	N/A	N/A	N/A
scaffold9_cov118	2402	2403	1	N/A	N/A	N/A
scaffold9_cov118	2405	2406	1	N/A	N/A	N/A
scaffold9_cov118	2452	2453	1	N/A	N/A	N/A
scaffold9_cov118	2461	2462	1	N/A	N/A	N/A
scaffold9_cov118	2509	2510	1	N/A	N/A	N/A
scaffold9_cov118	2514	2515	1	1	N/A	N/A
scaffold9_cov118	2517	2518	1	N/A	N/A	N/A
scaffold9_cov118	2596	2597	1	N/A	N/A	N/A
scaffold9_cov118	2614	2615	1	N/A	N/A	N/A


In [44]:
#substitute NA for 0
!head Pact_method_union_5x.bedgraph |\
sed 's/N\/A/0/g' 

chrom	start	end	ALL	WGBS	RRBS	MBD
scaffold100000_cov51	5	6	1	0	0	0
scaffold100000_cov51	23	24	1	1	0	0
scaffold100000_cov51	61	62	1	1	0	0
scaffold100000_cov51	69	70	1	1	0	0
scaffold100000_cov51	79	80	1	1	0	0
scaffold100000_cov51	110	111	1	1	1	0
scaffold100000_cov51	113	114	1	1	1	0
scaffold100000_cov51	206	207	1	1	1	0
scaffold100000_cov51	214	215	1	1	1	0


In [47]:
#combine fields 1-3 into 1 string
!cat Pact_method_union_5x.bedgraph |\
sed 's/N\/A/0/g' |\
awk -F"\t" '{print$1"_"$2"_"$3"\t"$4"\t"$5"\t"$6"\t"$7}' \
> Pact_method_union_5x_matrix.tsv

In [48]:
!head Pact_method_union_5x_matrix.tsv

chrom_start_end	ALL	WGBS	RRBS	MBD
scaffold100000_cov51_5_6	1	0	0	0
scaffold100000_cov51_23_24	1	1	0	0
scaffold100000_cov51_61_62	1	1	0	0
scaffold100000_cov51_69_70	1	1	0	0
scaffold100000_cov51_79_80	1	1	0	0
scaffold100000_cov51_110_111	1	1	1	0
scaffold100000_cov51_113_114	1	1	1	0
scaffold100000_cov51_206_207	1	1	1	0
scaffold100000_cov51_214_215	1	1	1	0


In [49]:
mkdir /Volumes/web/metacarcinus/FROGER_meth_compare/20200501

In [50]:
!rsync \
--progress \
--verbose \
--archive \
--exclude "gannet.fish.washington.edu" \
/Users/strigg/Documents/GitHub/Meth_Compare/data \
/Volumes/web/metacarcinus/FROGER_meth_compare/20200501


building file list ... 
30 files to consider
data/
data/MBD_5x.tc.bed
   117436711 100%   44.40MB/s    0:00:02 (xfer#1, to-check=28/30)
data/MBD_5x.tc.bedgraph
   124846389 100%   36.42MB/s    0:00:03 (xfer#2, to-check=27/30)
data/Meth1_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph
   189961287 100%   38.73MB/s    0:00:04 (xfer#3, to-check=26/30)
data/Meth1_R1_001_val_1_bismark_bt2_pe._5x.tab
   251446400 100%   33.25MB/s    0:00:07 (xfer#4, to-check=25/30)
data/Meth2_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph
   218602336 100%   36.93MB/s    0:00:05 (xfer#5, to-check=24/30)
data/Meth2_R1_001_val_1_bismark_bt2_pe._5x.tab
   289019583 100%   39.66MB/s    0:00:06 (xfer#6, to-check=23/30)
data/Meth3_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph
   201238522 100%   31.92MB/s    0:00:06 (xfer#7, to-check=22/30)
data/Meth3_R1_001_val_1_bismark_bt2_pe._5x.tab
   266285364 100%   43.11MB/s    0:00:05 (xfer#8, to-check=21/30)
data/Meth4_R1_001_val_1_bismark_bt2_pe._5x.tC.bedgraph
    63596292 100%

In [51]:
!wc -l Pact_method_union_5x_matrix.tsv

 9155621 Pact_method_union_5x_matrix.tsv


In [52]:
!head Pact_method_union_5x_matrix.tsv

chrom_start_end	ALL	WGBS	RRBS	MBD
scaffold100000_cov51_5_6	1	0	0	0
scaffold100000_cov51_23_24	1	1	0	0
scaffold100000_cov51_61_62	1	1	0	0
scaffold100000_cov51_69_70	1	1	0	0
scaffold100000_cov51_79_80	1	1	0	0
scaffold100000_cov51_110_111	1	1	1	0
scaffold100000_cov51_113_114	1	1	1	0
scaffold100000_cov51_206_207	1	1	1	0
scaffold100000_cov51_214_215	1	1	1	0
